mempool/src/mempool_interface_impl/pool.rs

```diff
@@ -219,7 +164,7 @@ impl<M> Mempool<M>
 where
     M: GetMemoryUsage + Send + std::marker::Sync,
 {
```

Comment:
This is a huge implementation block that can probably be broken into smaller pieces. But not in this PR. Once you merge this PR, you can do this next.

Reply:
Opened an issue for this.